File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        int highestCandyCountBeforeExtras = candies[0];
        
        for (int i = 1; i < candies.length; i++) {
            if (candies[i] > highestCandyCountBeforeExtras) {
                highestCandyCountBeforeExtras = candies[i];
            }
        }
        
        List<Boolean> greatestOrNot = new ArrayList<>();
        
        for (int candy : candies) {
            if (candy + extraCandies >= highestCandyCountBeforeExtras) {
                greatestOrNot.add(true);
            } else {
                greatestOrNot.add(false);
            }
        }
        
        return greatestOrNot;
    }
}

File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.js
/**
 * @param {number[]} candies
 * @param {number} extraCandies
 * @return {boolean[]}
 */
var kidsWithCandies = function(candies, extraCandies) {
    const maxCandies = Math.max(...candies);
    const bools = candies.map(candy => candy + extraCandies >= maxCandies);
    return bools;
//     var maxCandies = candies[0];
//     for (let i = 1; i <= candies.length; i++) {
//         if (candies[i] > maxCandies) {
//             maxCandies = candies[i];
//         }
//     }
//     var bools = [];
//     candies.forEach(function(candy) {
//        if (candy + extraCandies >= maxCandies) {
//            bools.push(true);
//        } else {
//            bools.push(false);
//        }
//     });
    
//     return bools;
};